backend/common/db/core.py

```
@@ -19,16 +21,60 @@
 engine = create_engine(settings.database_url, **_engine_kwargs)
+
+
+# Connection pool monitoring for production/demo environments
+@event.listens_for(Engine, "connect")
+def log_connect(dbapi_conn, connection_record):
+    """Log when new database connections are established."""
+    logger.info("Database connection pool: New connection established")
+
+
+@event.listens_for(Engine, "checkout")
+def log_checkout(dbapi_conn, connection_record, connection_proxy):
```

Ruff (0.14.7)

Unused function argument: `dbapi_conn`

(ARG001)